src/components/MapView/Analyser/index.tsx

```diff
@@ -277,37 +342,67 @@ function Analyser({ extent, classes }: AnalyserProps) {
       return;
     } // hasn't been calculated yet
 
-    if (!selectedDate) {
-      throw new Error('Date must be given to run analysis');
+    if (!selectedHazardLayer) {
+      throw new Error('Hazard layer should be selected to run analysis');
     }
 
-    if (!hazardLayerId || !baselineLayerId) {
-      throw new Error('Layer should be selected to run analysis');
-    }
+    if (hazardDataType === GeometryType.Polygon) {
+      if (!startDate) {
+        throw new Error('Date Range must be given to run analysis');
+      }
+      if (!endDate) {
+        throw new Error('Date Range must be given to run analysis');
+      }
+      if (!adminLevelLayer || !adminLevelLayerData) {
+        // technically we can't get here because the run analaysis button
+        // is disabled while the admin level data loads
+        // but we have to put this in so the typescript compiler
+        // doesn't throw an error when we try to access the data
+        // property of adminLevelLayerData
+        throw new Error('Admin level data is still loading');
+      }
 
-    const selectedHazardLayer = LayerDefinitions[
-      hazardLayerId
-    ] as WMSLayerProps;
-    const selectedBaselineLayer = LayerDefinitions[
-      baselineLayerId
-    ] as AdminLevelDataLayerProps;
+      const params: PolygonAnalysisDispatchParams = {
+        hazardLayer: selectedHazardLayer,
+        adminLevel,
+        adminLevelLayer,
+        adminLevelData: adminLevelLayerData.data,
+        startDate,
+        endDate,
+        extent,
+      };
+      activateUniqueBoundary(adminLevelLayer);
+      dispatch(requestAndStorePolygonAnalysis(params));
+    } else {
+      if (!selectedDate) {
+        throw new Error('Date must be given to run analysis');
+      }
+
+      if (!baselineLayerId) {
+        throw new Error('Baseline layer should be selected to run analysis');
+      }
+
+      const selectedBaselineLayer = LayerDefinitions[
+        baselineLayerId
+      ] as AdminLevelDataLayerProps;
+
+      activateUniqueBoundary();
 
-    activateUniqueBoundary();
-
-    const params: AnalysisDispatchParams = {
-      hazardLayer: selectedHazardLayer,
-      baselineLayer: selectedBaselineLayer,
-      date: selectedDate,
-      statistic,
-      extent,
-      threshold: {
-        above: parseFloat(aboveThreshold) || undefined,
-        below: parseFloat(belowThreshold) || undefined,
-      },
-      isExposure: false,
-    };
-
-    dispatch(requestAndStoreAnalysis(params));
+      const params: AnalysisDispatchParams = {
+        hazardLayer: selectedHazardLayer,
+        baselineLayer: selectedBaselineLayer,
+        date: selectedDate,
+        statistic,
+        extent,
+        threshold: {
+          above: parseFloat(aboveThreshold) || undefined,
+          below: parseFloat(belowThreshold) || undefined,
+        },
+        isExposure: false,
```

Comment:
Nothing seems to use this flag? Should we remove it from the type?

Reply:
And from the different places where it is set?

Reply:
agreed. done with a140d23